llvm/unittests/IR/PatternMatch.cpp

```diff
@@ -625,39 +625,49 @@ TEST_F(PatternMatchTest, CheckedInt) {
     APInt APVal(8, Val);
     Constant *C = ConstantInt::get(I8Ty, Val);
 
+    CRes = nullptr;
     Res = nullptr;
     EXPECT_TRUE(m_CheckedInt(CheckTrue).match(C));
-    EXPECT_TRUE(m_CheckedInt(Res, CheckTrue).match(C));
+    EXPECT_TRUE(m_CheckedInt(CRes, CheckTrue).match(C));
+    EXPECT_NE(CRes, nullptr);
+    EXPECT_TRUE(match(CRes, m_APIntAllowPoison(Res)));
```

Comment:
```suggestion
    EXPECT_EQ(CRes, C);
```
I think we should be checking the constant directly, not the APInt it contains.

Reply:
Sure, how it is now was mostly based on the simplest changes...